### PR TITLE
Fix race condition in CompareDataFrameStage

### PR DESCRIPTION
**Pull request by dagardner-nv**
## Description
* Fixes intermittent segfault in `CompareDataFrameStage` and `ValidationStage`

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/nv-morpheus/Morpheus/blob/main/docs/source/developer_guide/contributing.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: morpheus/stages/output/compare_dataframe_stage.py [hunk 1]
```diff
@@ -107,13 +107,15 @@ def get_results(self, clear=True) -> dict:
         if self._reset_index:
             combined_df.reset_index(inplace=True)
 
+        retults = compare_df.compare_df(self._compare_df,
```
Comment (drobison00) [inline morpheus/stages/output/compare_dataframe_stage.py:110]:
```suggestion
        results = compare_df.compare_df(self._compare_df,
```

File: morpheus/stages/output/compare_dataframe_stage.py [hunk 2]
```diff
@@ -107,13 +107,15 @@ def get_results(self, clear=True) -> dict:
...
             self.clear()
 
-        return compare_df.compare_df(self._compare_df,
-                                     combined_df,
-                                     include_columns=self._include_columns,
-                                     exclude_columns=self._exclude_columns,
-                                     replace_idx=self._index_col,
-                                     abs_tol=self._abs_tol,
-                                     rel_tol=self._rel_tol)
+        return retults
```
Comment (drobison00) [inline morpheus/stages/output/compare_dataframe_stage.py:121]:
```suggestion
        return results
```

**Review by drobison00 — APPROVED** [review]

**Review by cwharris — APPROVED** [review]
I wonder if `get_messages` should return a copy of the message array rather than an instance of it...

**Comment by dagardner-nv** [comment]
> I wonder if `get_messages` should return a copy of the message array rather than an instance of it...

I considered that, but decided against it since:
1) It would come at a higher cost.
2) Calling `clear` should be self descriptive.

**Pull request merged**